transport/TransportFactory.cpp

```diff
@@ -484,6 +488,36 @@ class TransportFactoryImpl final : public TransportFactory,
         endpoint->stop(this);
     }
 
+    void registerProbeServer(ProbeServer& server) override
+    {
+        for (auto& endpoint : _tcpServerEndpoints)
+        {
+            server.registerEndpoint(*endpoint);
+        }
+
+        auto udpProbeEndpoint = _sharedEndpoints[0]; // config.ice.singlePort
```

Comment:
register in all interfaces on _sharedEndpoints. On current state we will have only one but we already discuss in having a range of ports as we are reaching the limits the packets we can send in one single port on load tests

Reply:
In the end if the port range it big, we will have a lot of ICE candidates as well, but we can filter some of them and only provide 2 or 3 of them

Reply:
I choose one UDP endpoint because of @olofkallander line in the ticket "think one of the shared UDP ports we have could be used to handle this". I can change it to add listener (ProbeServer) to all the UDP endpoints.

Reply:
Now I think it'd be better to simply pass all the TcpServer and UDP endpoints from TransportFactory to ProbeServer in register method and let the server decide which ones to use for probing.

```
register(ProbeServer& server)
{
    server.attach(_sharedEndpoints, _tcpServerEndpoints);
}
```

Reply:
Yeah, you can do like that but keep in mind probably we want to have all endpoints for the same port. Imagine you have 2 NIC. 1st IP `10.0.0.1` and 2nd `10.0.0.2`. And we have port range from 10000-10005.

So even if we want to select only one port (let's say port 10000) you should include both endpoints `10.0.0.1:1000` and `10.0.0.2:1000` because we don't know which IP is reachable from client. That perhaps is more difficult to control inside on `ProbeServer`. `TransportFactory` has a vector with another vector. First vector is about the port and the second vector is all network interfaces where that port is bound.

I think if one day we will have more than 1 port available, maybe we should provide candidates for at least for 2 different ports, so in a load situation where 1 port is very loaded and some packets will be lost, client can use a second port to probe

Reply:
Yes, here _sharedEndpoints is vector of vectors

```
std::vector<Endpoints>
typedef std::vector<std::shared_ptr<Endpoint>> Endpoints;
```

Also as I understand, we need to **listen** on all the interfaces, but in API we report only public IPv4 and optionally IPv6 addresses. So ProbeServer would only need to know port of specific endpoint and retrieve public IPs from ICE config.